### PR TITLE
[android-auto] Refactoring of PlaceScreen and NavigationScreen

### PR DESCRIPTION
**Pull request by AndrewShkrob**
* Simplified logic of PlaceScreen
* Reduced minSdkVersion: 23 -> 21
* Fixes: https://github.com/organicmaps/organicmaps/issues/5759

Demo:

https://github.com/organicmaps/organicmaps/assets/10351358/75d2feca-c48e-4fe7-9f93-a800206b2a62

### CODE REVIEW

**Review by Jean-BaptisteC — CHANGES_REQUESTED** [review]
❌ Pixel 6 - Android 13

- Select POI on the map from AA
- Switch to phone -> Layout is wrong (On my Pixel 6, routing icons are behind camera)
- We have the same problem when you switch to phone when routing is started

On AA, if I select POI and after another POI between switch, search bar + categories appear during 1 second, is it possible to fix?

**Comment by Jean-BaptisteC** [comment]
Why size app been reduced in this version?

**Comment by AndrewShkrob** [comment]
> ❌ Pixel 6 - Android 13

These bugs are related to the main app. I am aware of them and will address them later. The current PR only fixes the behavior on AA.

> On AA, if I select POI and after another POI between switch, search bar + categories appear during 1 second, is it possible to fix?

No, this issue is caused by the limitation of AA in terms of screen stack size and the inappropriate implementation of this logic within the car library. Our current solution addresses it, although it results in a "small" disadvantage of screen flickering. This matter was previously discussed here: https://github.com/organicmaps/organicmaps/pull/5045

**Review by rtsisyk — CHANGES_REQUESTED** [review]
In my opinion, this is a very controversial refactoring. 

There are two screens that need to work with app.organicmaps.routing.RoutingController (already existing in the code, please don't be confused with app.organicmaps.car.NavigationController added by this patch) - PlacePage and NavigationScreen. PlacePage is used to build and show the route. NavigationScreen is for actual navigation logic. Each screen has a different purpose and implements a different subset of app.organicmaps.routing.RoutingController.Container callbacks. 

What was the point of creating one more abstraction level on the top of the existing RoutingController to merge unrelated methods from PlacePage and NavigationScreen? Why can't we keep implementation of callbacks that are necessary for building the route in PlaceScreen and callbacks needed for navigation in NavigationScreen? There may be a slight overlap in the implementation, but it is not critical. The presence of  `screen instanceof PlaceScreen` and `screen instanceof NavigtionScreen` is a good indication that this new app.organicmaps.car.NavigationController is leaking.

**Comment by AndrewShkrob** [comment]
@rtsisyk PTAL
@Jean-BaptisteC could you please retest? I've updated the logic, but the behavior should remain the same.

File: android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -902,7 +901,7 @@
 
     <meta-data
         android:name="androidx.car.app.minCarApiLevel"
-        android:value="1"/>
+        android:value="6"/>
```
[the remark this answers is not in the document]
Comment (AndrewShkrob) [reply android/app/src/main/AndroidManifest.xml:904]: We already use an API that is supported only from CarApiLevel >= 6. If we define the min level explicitly in the manifest, our app won't be visible on hosts with lower API. If we don't define it explicitly, the app will fail.

I still don't know how it works. Does the host need to support this CarApiLevel, or is it only related to the Android Auto application? If this value relates to the host, then we're in trouble.

By the way, navigation is supported only from CarApiLevel >= 4.

If there will be some users with hosts that don't support our app, then we can think about supporting older APIs (it's possible).

**Review by Jean-BaptisteC — APPROVED** [review]
✅ Pixel 6 - Android 13 + AA Emulator

Why when I select a place we don't have a progress bar on button start to start routing automatically?

File: android/app/src/main/AndroidManifest.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (biodranik) [reply android/app/src/main/AndroidManifest.xml:904]: 1. What API from level 6 are we using?
2. How many cars are supporting API 6 vs API 4?
3. What does it mean "Android Auto without navigation support" on API 1, 2, and 3? What was the point of it?

[the remark this answers is not in the document]
Comment (rtsisyk) [reply android/app/src/main/AndroidManifest.xml:904]: This change is not forced by changed added by this PR. Existing code probably already violates API < 6 contract. Let's review things and evaluate what API level should be supported  in https://github.com/organicmaps/organicmaps/issues/6105. I added it to the list of blockers.

**Comment by AndrewShkrob** [comment]
> Why when I select a place we don't have a progress bar on button start to start routing automatically?

They broke everything again :( 
I think it's better to leave it this way until it's fixed.

| Before | Now |
|--------|--------|
| telegram-cloud-photo-size-2-5470138097455976427-x | telegram-cloud-photo-size-2-5470138097455976429-x |
| telegram-cloud-photo-size-2-5470138097455976428-x | telegram-cloud-photo-size-2-5470138097455976430-x |

**Pull request closed without merging**